### DISCUSSION TITLE
[Nakamoto] Best Practices in Consuming Stacks Node Blocks

### DISCUSSION THREAD

**Discussion by AshtonStephens** [stacksgov/sips#164] [Ideas]
This is a research discussion to determine the best practices when consuming blocks from the stacks node so as to not affect the stacks node's performance and other such details.

**Comment by AshtonStephens** [comment]
Q: @rafaelcr What would be a reasonable assumption in that case related to the event RPCs that an API gets from the stacks node. If the API gets low for any reason, is there any circumstance in which the API could negatively impact the performance of the node? 

A: @jcnelson An API should take blocks off the event observer as soon as it can and store them locally. Nodes shouldn't be blocked on processing them to prevent impact to the 5s block time, and should also have a backlog stored locally.

**Reply by rafaelcr** [reply]
A follow-up question is: assuming there's an API crash that makes it miss some block events while it restarts, is there a way it can retrieve those past block events from the Stacks node so it can catch up to chain tip?
